firebase-ai/src/main/kotlin/com/google/firebase/ai/type/LiveServerMessage.kt

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.ai.type
+
+import kotlinx.serialization.DeserializationStrategy
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.JsonContentPolymorphicSerializer
+import kotlinx.serialization.json.JsonElement
+import kotlinx.serialization.json.JsonNull
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.jsonObject
+
+/**
+ * Parent interface for responses from the model during live interactions.
+ *
+ * @see LiveServerContent
+ * @see LiveServerToolCall
+ * @see LiveServerToolCallCancellation
+ * @see LiveServerSetupComplete
+ */
+@PublicPreviewAPI public interface LiveServerMessage
+
+/**
+ * Incremental server update generated by the model in response to client messages.
+ *
+ * Content is generated as quickly as possible, and not in realtime. You may choose to buffer and
+ * play it out in realtime.
+ */
+@PublicPreviewAPI
+public class LiveServerContent(
+  /**
+   * The content that the model has generated as part of the current conversation with the user.
+   *
+   * This can be `null` if there is no content.
+   */
+  public val content: Content?,
+
+  /**
```

Comment:
I've tried this in Android Studio but couldn't get a definitive answer. While doc show file for the particular field, do they show in the constructor? should the constructor be public or internal?

Reply:
The docs _do_ show up in the constructor, if that's what you're asking. _Maybe_ it should be internal, but we also have the constructor on `GenerateContentResponse` as public- so I was just aligning with that. The only practical use-case I can think of for them being public is consumers wanting to write tests for their interop layer, or their logic directly if they don't have an interop layer.

Would you rather we make both of their constructors internal instead?

Reply:
That would be a bigger breaking change that we can hold for later.